demos/tabs/ajax.html

```diff
@@ -8,10 +8,27 @@
 	<link rel="stylesheet" href="../demos.css">
 	<script src="../../external/requirejs/require.js"></script>
 	<script src="../bootstrap.js">
+		var jqAjax = $.ajax;
+		$.ajax = function( options ) {
+			var jqXHR = jqAjax.apply( this, arguments );
+			if ( !/slow.html/.test( options && options.url ) ) {
+				return jqXHR;
+			}
+			return jqXHR
+				.then(function( result ) {
+					return $.Deferred( function( d ) {
+						setTimeout( function() {
+							d.resolve( result );
+						}, 1000 );
+					});
+				})
+				.promise({ abort: jqXHR.abort });
+		}
```

Comment:
Interesting. I was looking into ways to potentially handle this case via prefilters/transports but there's no clear way as I understand. A transport can either bail out or handle the request, there's no "do your own thing and pass the control to the next one" like middlewares typically behave.

Reply:
Indeed, I did use middlewhere for the other demos, but I was unable to find something that simpler than a monkey-patch for this use case.